### PR TITLE
Update .importlinter to enforce Mesh -> NN, and add Optim

### PR DESCRIPTION
**Pull request by coreyjadams**
This adds mesh and optim both to the import linter chain.

Mesh itself could use a breakdown of it's subcomponents.  That can come in a separate PR from @peterdsharpe

# PhysicsNeMo Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [x] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

**Review by peterdsharpe — CHANGES_REQUESTED** [review]
The current state of this reorganization looks acceptable to me, both in terms of `optim` and `mesh`! Thanks @coreyjadams for carrying the torch forward on this one!

However before merging we should resolve:

- It looks like Greptile is correct here that `nn` is importing from `mesh`. From what I can tell, `nn.functional.sdf` should not be importing `lumpy_ball` from `mesh` - this is just example code. I can envision a case where the `nn` **test suite** imports `lumpy_ball`, but I'm not sure this belongs in core. Perhaps @loliverhennigh can discuss more though (or add fixes, if this is deemed not needed)? I might not be fully understanding some reason why this needs to be in core, not in the tests.

- I would propose that for now we make `nn : mesh` siblings, which cannot mutually import from each other - since currently, they're almost totally independent (with the exception of the issue above ^ ). Then, down the road (e.g., when Oliver's gradient perf improvement comes in), we can change this as part of that PR. This preserves optionality as long as possible - if there's no current dependency direction, perhaps it's best to keep it this way until we decide later on. I think @ktangsali and @loliverhennigh and I are having a design meeting next week about this - @coreyjadams I'll loop you on too if you're interested.

**Comment by peterdsharpe** [comment]
@loliverhennigh, whenever you have a moment - would you mind pushing a commit to this branch to make the `nn` -> `mesh` dependency one-way (by refactoring the SDF `make_inputs` example)?

File: .importlinter [hunk 1]
```diff
@@ -13,7 +13,8 @@ containers=
 layers = 
    experimental
    active_learning : diffusion
-   models : datapipes : metrics : domain_parallel
+   models : datapipes : metrics : domain_parallel : optim
+   mesh
    nn
```
[the remark this answers is not in the document]
Comment (peterdsharpe) [reply .importlinter:18]: Fixed by refactoring sdf.
